hax/hax/util.py

```diff
@@ -761,6 +767,29 @@ def set_sdev_state(self,
             value['state'] = state
             self.kv.kv_put(sdev['Key'], json.dumps(value))
 
+    @repeat_if_fails()
+    def get_sdev_state(self, obj_t: ObjT, fidk: int) -> int:
+        if obj_t.name == ObjT.DRIVE.name:
+            drive_fid = create_drive_fid(fidk)
+            sdev_fid = self.drive_to_sdev_fid(drive_fid)
+        else:
+            sdev_fid = create_sdev_fid(fidk)
+        LOG.debug('Getting state for sdev=%s', str(sdev_fid))
+        sdev_items = self.kv.kv_get('m0conf/nodes', recurse=True)
+        regex = re.compile(
+            f'^m0conf\\/.*\\/sdevs\\/{sdev_fid}$')
+        for sdev in sdev_items:
+            match_result = re.match(regex, sdev['Key'])
+            if not match_result:
+                continue
+            state = json.loads(sdev['Value'])
+            if str(state).lower() in ('unknown', 'offline', 'failed'):
```

Comment:
I have a doubt, whether `sdev['Value']` will return `{ "path": "<sdev path>", "state": "<HA state>" }` or only the state ?
Like in **set_sdev_state()** -
```
value = json.loads(sdev['Value'])
value['state'] == 'failed' 
```
Please could you confirm.

Reply:
Indeed, good catch. m0client client join still fails with this patch because controller, node and enclosure failures are not part of the nvec reply, that will be fixed in another ticket, but tested with ioservice restart with a stopped node and ioservice does get correct device statuses in nvec reply.